### ISSUE TITLE
TS service validator incompatible with the latest @types/feathersjs__feathers

### ISSUE THREAD

**Issue by hakandilek** [feathers-plus/generator-feathers-plus#49]
I've generated a TS based app with a service but the `services/<service>/<service>.validate.ts` file reports `TSError`s between lines 66 and 68:

![image](https://user-images.githubusercontent.com/1072473/45888151-ab95f400-bdbd-11e8-8c51-0db880b587e7.png)

Recently updated [@types/feathersjs__feathers](https://github.com/DefinitelyTyped/DefinitelyTyped/blame/master/types/feathersjs__feathers/index.d.ts) expects a lot more new properties on `HookContext`:

![image](https://user-images.githubusercontent.com/1072473/45888388-37a81b80-bdbe-11e8-86b0-413b2cee5992.png)

**Comment by eddyystop** [comment]
@j2l4e ^^^

**Comment by j2L4e** [comment]
Oops :/
The typings were changed to be in line with how context objects constructed by feathers look according to the docs.
We'll need to discuss that in general.
Wie can either (for now) add `as any` after the constructed context object or add the missing properties.
I'm out of town right now, will be back sunday night, then i'll try to find a proper solutiom for this.

**Comment by j2L4e** [comment]
```
    if (method === 'create') { validateCreate(options)({ type: 'before', method: 'create', data } as any); }
    if (method === 'update') { validateCreate(options)({ type: 'before', method: 'update', data } as any); }
    if (method === 'patch') { validateCreate(options)({ type: 'before', method: 'patch', data } as any); }
```

**Comment by eddyystop** [comment]
The tests are totally worth it. I would have little confidence in the generator without them.

However its still annoying that its takes 20 times as long to change the tests than to change the generator itself. LOL

Published in f+/cli v0.7.32 using generator v0.6.31

**Comment by eddyystop** [comment]
Does this resolve the duplicate issue? If so, could you close that j2l4e?

**Comment by eddyystop** [comment]
I closed #52.
